services/graph/pkg/identity/odata.go

```diff
@@ -37,5 +41,9 @@ func GetSearchValues(req *godata.GoDataQuery) (string, error) {
 		return "", godata.NotImplementedError("complex search queries are not supported")
 	}
 
-	return req.Search.Tree.Token.Value, nil
+	searchValue := req.Search.Tree.Token.Value
+	if strings.HasPrefix(searchValue, "\"") && strings.HasSuffix(searchValue, "\"") {
+		searchValue = strings.Trim(searchValue, "\"")
+	}
```

Comment:
Nitpick: I guess in this case you can just call `strings.Trim` unconditionally. There is no need for the `HasPrefix` and `HasSuffix` calls. Or am I missing something?

The OData parser will already choke on strings that only contain a single double quote.

Reply:
> The OData parser will already choke on strings that only contain a single double quote.

I thought it was still possible. I'll remove the condition.

Reply:
How are we dealing the same with single quotes?

Reply:
I think single quotes have a completely different meaning here and are handled by the OData parser.  In this case, as far as I understand the odata specs, using single quotes here won't even work. (our parsers doesn't allow it here at least https://github.com/CiscoM31/godata)

Reply:
does a _user_ (the one who is on the keyboard...) know this difference...?

Reply:
The user shouldn't need to worry about quoting at all.

The UI/client which is constructing the query should know the difference, yes.